lib/messages.js

```diff
@@ -3,6 +3,34 @@ var _ = require('underscore'),
     utils = require('./utils'),
     logger = require('./logger');
 
+function extendedTemplateContext(doc, extras) {
```

Comment:
I considered moving the documentation from the function below, but I figured since the function is basically just a bunch of `if (extras.BLAH)` statements it was obvious enough?